pkg/plugins/manager/loader/finder/local.go

```diff
@@ -81,15 +88,23 @@ func (l *Local) Find(ctx context.Context, src plugins.PluginSource) ([]*plugins.
 
 	var res = make(map[string]*plugins.FoundBundle)
 	for pluginDir, data := range foundPlugins {
-		files, err := collectFilesWithin(pluginDir)
-		if err != nil {
-			return nil, err
+		var pluginFs plugins.FS
+		pluginFs = newDefaultLocalFS(pluginDir)
+		if !l.devMode {
+			// In prod, tighten up security by allowing access only to the files present up to this point.
+			// Any new file "sneaked in" won't be allowed and will acts as if the file did not exist.
+			// Build the allow-list by calling Files() once
+			fsFiles, err := pluginFs.Files()
+			if err != nil {
+				return nil, err
+			}
+			// Wrap FS with allow list
+			pluginFs = plugins.NewAllowListFS(pluginFs, fsFiles...)
```

Comment:
The idea was to have `AllowListFS` to act as an additioanl "filter" in front of any `plugins.FS`, so you could technically give it an underlying `LocalFS` (which can access a whole folder on the local filesystem), but by wrapping it into an `AllowListFS` you could allow access to a subset of files/folders exposed by the `LocalFS`.

I think we can either:

1. Add a `NewAutoAllowListFS` or something similar, which will build the allow-list based on the result of `Files()` on creation, and it will keep working in this case.

```go
func NewAutoAllowListFS(fs FS) (AllowListFS, error) {
	fsFiles, err := pluginFs.Files()
	if err != nil {
		return nil, err
	}
	return NewAllowListFS(fs, fsFiles...)
}
```

2. ...or change `NewAllowListFS` to always have that behaviour (removing the possibility to act as a "filter", but just avoiding allowing access to files added after initialization).

Any preferences or alternative solutions? 2 is fine for this case, 1 is more flexible but not sure if we need such flexibility now, as we only have LocalFS and CDN (which acts a bit differently anyways).

Reply:
Yeah I think option 2 makes sense for now 👍

Reply:
Thanks, I'll add that shortly :smile: